language-design.md

```diff
@@ -26,32 +30,32 @@ you know a poorly designed language? What are the symptoms?
-class?
+How might the themes of _Growing a Language_ relate to ideas from the Fowler reading?
 
 **Response**
 
----
+More obviously, the idea of limited expressiveness that Fowler discusses is very related to _Growing a Language_ and the ways that Steele uses limited expressiveness to make his points. Of using only one syllable words, unless defined first, he says "in truth, the words of one syllable form quite a rich vocabulary, with which you can say many things." [Steele, 1995] He is able to use a small language and express many things quite fluently, or at least the bare minimum things that he needs - note that Steele doesn't define anything he doesn't use. Similarly, Fowler characterizes DSls with limited expressiveness, and that it doesn't have extraneous features, just what it needs to support its domain. [Fowler, 2010] There are a few other design principles Fowler and Steele both highlight. First, the importance of user input. In Fowler, he talks about the benefits of DSLs as (and needing) communication with domain experts - they are the ones with the specific knowledge about how to make the DSL most helpful. [Fowler, 2010] This is echoed in Steele's discussion of building things as a bazaar, and gathering lots of helping hands and inputs. [Steele, 1995] That also builds into another point Fowler makes about improving development productivity. Both readings thematically are trying to improve an experience and the tools one uses, in order to also improve what someone can do with a programming language.
 
 **Question**
 
 In what way is an API a language?
 
 **Response**
 
----
+An API a language because it is a set of rules for how programs can talk to each other. If a programming language is a set of rules for communicating with a computer to solve a problem, then using an API is the same, it's just a subset of rules presented in a different way. This is why the article on designing a good API is important, because it needs to be fluent and have overlapping principles with designing a general purpose language, like good names and documentation. [Bloch, 2006]
 
 **Question**
 
 What does the post on grayscale tell us about the process of API design?
 
 **Response**
 
----
+It tells us that there are lots of ways to to name functions that do the same thing with the same result, and so user input is important in that process to make sure that most people find it intuitive. That being said, because there are so many "correct" ways to do things, and everyone has different ways of thinking about problems and different ideas (like how to quantify the color gray) that not everyone is going to be happy. I feel like it is following Bloch's ideas for good API design, because it's giving attention to the importance of names, getting input, and trying to be as least surprising as possible. [Bloch, 2006] Essentially, the API design process as modelled here is purposeful and public so that it is as intuitive and effectively designed as possible.
```

Comment:
You can't make everyone happy, so you should instead try to make the least number of people miserable.